hooks/README.md

```diff
@@ -0,0 +1,98 @@
+# Camio Hooks
+
+## Introduction
+
+Camio provides an image processing pipeline. Videos are uploaded to the Camio cloud from
+a phone or an RTSP camera or a Camio Box. The Camio image pipeline provides a labeling 
+service but also allows you to register your own labeling hooks. A hook is defined as a custom web service that can compute labels from images.
+
+As videos are uploaded to Camio, they are broken into events,
+which are classified as boring, motion, and interesting. Interesting events are pre-classified by the Camio service, labeled, and a significative images are extracted from the videos in the event. If you have registsred a hook, and the event labels match your hook filter, the images beloging to the event are POSTed to your hook URL in the form of a JSON payload. Your hook URL should record the JSON payload, decoded it, extract the images, label them, and then POST the labels back to Camio at the callback_url provided by the initial hook call. The new labels are added to the event.
+
+Remember your hook labels events, not images, or videos. One event may contain mutiple images and multiple videos. The same video can span multiple consecutive events. At this time we only send to your hook preprocessed images. Also, you only label events that Camio has alreday pre-filtered for you as interesting.
```

Comment:
"Your hook labels an entire Event (not just a single image or video) based on the images and/or video in that Event."
TODO(massimo) - hooks need a way to POST the exact correlation between image and labels even if we index the labels for the entire event. We should specify a json structure that supports timestamp correlation.